(SDI-2092) Update README Getting started section

nanliu opened the pull request:
* use package installation and service commands when possible
* use psutil instead of trivial plugin example
* ensure task works on both linux and MacOS

IRCody commented on README.md:
````diff
@@ -86,164 +86,166 @@ Some additionally important notes about how snap works:
 ```
 
-Or tap into the data that Snap is collecting using the Task ID to watch the task (note: your Task ID will be different):
+Or directly tap into the data stream that Snap is collecting using `snapctl task watch` (
````
Should we be explicit about this being the task ID here? i.e. `snapctl task watch <task-id>`

nanliu replied:
Updated.

IRCody commented on README.md:
````diff
@@ -86,164 +86,166 @@ Some additionally important notes about how snap works:
-```
-
-Nice work - you're all done with this example. You ran `snapd` manually, so stopping the daemon process will stop any running tasks and unload any plugins we loaded.
+Nice work - you're all done with this example. You started `snap-telemetry` service earlier, so use the appropriate service command to stop the daemon, or run the following command if you ran `snapd` manually:
````
I feel like the condition is flipped in this sentence.  Maybe something like?

```
If you started `snap-telemetry` as a service earlier, use the appropriate service command to stop the daemon. If you ran `snapd` manually use:
```

Or alternatively something like:
```
If you started `snap-telementry` service earlier, use the appropriate service command to stop the daemon, or run the following command if you ran `snapd` manually:
```

nanliu replied:
I changed this section a bit more to list all three possible commands. Let me know if it's clear.

IRCody replied:
LGTM

nanliu commented:
@IRCody, we released psutil version 8, and I provided the darwin binary, so this PR can be merged when we pass review.

katarzyna-z commented on README.md:
```diff
@@ -86,163 +85,166 @@ Some additionally important notes about how snap works:
 ### System Requirements
-Snap deploys as a binary, which makes requirements quite simple. We've tested on a subset of Linux and OS X versions.
+
+Snap does not have external dependencies since it's compiled into a staticly linked binary. At this time, Snap only runs on MacOS and Linux. We generate RPM/Deb packages for Linux and pkgs for MacOS X. We also provide snapd/snapctl binaries for testing the latest build.
```
It seems that that there are typos and it should be "into a **statically** linked binary" instead of "into a **staticly** linked binary".

katarzyna-z commented on README.md:
````diff
@@ -86,163 +85,166 @@ Some additionally important notes about how snap works:
-See the data that is being published to the file:
+This starts a task collecting metrics via psutil, then publishes the data to a file. To see the data published to the file (CTRL+C to exit):
 ```
 $ tail -f /tmp/snap_published_mock_file.log
````
in exemplary task manifest the file is "psutil_metrics.log" so it seems that here the file name should be corrected

katarzyna-z commented on README.md:
````diff
@@ -86,163 +85,166 @@ Some additionally important notes about how snap works:
 ```
-$ cd ~/snap
-$ snapctl task create -t mock-file.yaml
+$ curl https://raw.githubusercontent.com/intelsdi-x/snap/master/examples/tasks/psutil-file.yaml -o /tmp/psutil-file.yaml
````
Here the exemplary task manifest is downloaded from e**xamples/tasks/** directory but you've added exemplary task manifest to **examples/** directory. Please check it.

nanliu replied:
I moved the task into the tasks subfolder.

kjlyon commented on README.md:
```diff
@@ -86,163 +85,166 @@ Some additionally important notes about how snap works:
 ### Installation
 
-You can get the pre-built binaries for your OS and architecture at Snap's [GitHub Releases](https://github.com/intelsdi-x/snap/releases) page. This isn't the comprehensive list of plugins. Right now, snap only supports Linux and OS X (Darwin).
+You can obtain Linux RPM/Deb packages from [Snap's packagecloud.io respository](https://packagecloud.io/intelsdi-x/snap). Snap binaries in tar gzip and MacOS pkg are available at Snap's [GitHub release page](https://github.com/intelsdi-x/snap/releases).
```
Do you need commas between tar, gzip, and MacOS?

kjlyon commented on README.md:
```diff
@@ -86,163 +85,166 @@ Some additionally important notes about how snap works:
-This will bring up a Snap agent without requiring plugin signing (trust-level 0) and set the logging level to debug (log level 1). Snap's REST API will be listening on port 8181. To learn more about the snap agent and how to use it look at [SNAPD.md](docs/SNAPD.md) and/or run `snapd -h`.
 
-Snap can also be run in a clustered mode called `tribe`. Checkout the [tribe documentation](docs/TRIBE.md) for more info.
+By default Snap daemon will be running standalone mode and listening on port 8181. To enable gossip mode, checkout the [tribe documentation](docs/TRIBE.md). For additional configuration options such as plugin signing and port configuration see [snapd documentation](docs/SNAPD.md).
```
Should add an `in` to be `running in standalone mode...`

nanliu commented:
@kjlyon, I changed the command to run in background and added instructions on where to find the log files so it's more consistent across all the different processes. I also changed tar gz to `tar.gz` and pkg to `.pkg` so it's a bit clearer we are referring to file extensions.

kjlyon commented:
LGTM

nanliu commented on examples/tasks/psutil-file.yaml:
```diff
@@ -0,0 +1,17 @@
+---
```
Yep, added.

nanliu commented:
Added addition notes about Ubuntu 16.04.1 to address and fix #1294.

jcooklin commented on README.md:
```diff
@@ -86,163 +85,175 @@ Some additionally important notes about how snap works:
 ### System Requirements
-Snap deploys as a binary, which makes requirements quite simple. We've tested on a subset of Linux and OS X versions.
+
+Snap does not have external dependencies since it's compiled into a staticly linked binary. At this time, Snap only runs on MacOS and Linux. We generate RPM/Deb packages for Linux and pkgs for MacOS X. We also provide snapd/snapctl binaries for testing the latest build.
```
Snap can run on Windows.  

~~At this time, Snap only runs on MacOS and Linux.~~
At this time, we build and package snap for Linux and MacOS X.

nanliu replied:
Removed

jcooklin commented on README.md:
````diff
@@ -86,163 +85,175 @@ Some additionally important notes about how snap works:
-```
 
-and then start the task:
+Tasks can be created directly via `snapctl`, but due to length of task it makes more sense to load from a `Task Manifest` written in JSON or YAML format.
````
"Tasks can be created directly via `snapctl`"

Adding the ability to generate a well-formed task manifest is a feature that should eventually be added but doesn't currently exist.

nanliu commented:
This is ready for review again.
